### PR TITLE
feat: support for currentcolor

### PR DESCRIPTION
**Pull request by LuciNyan**
### Description
Support for currentColor.

Before the `value` was passed to `css-to-react-native` for use, I rudely replaced `currentcolor` with the actual color. It looks like the `border` and `background` are working fine.

image

### CODE REVIEW

**Comment by LuciNyan** [comment]
Hi @jueungrace! you've done a lot of research on this, and I wonder if you see any risks. 🤔

**Comment by jueungrace** [comment]
> any risks. 🤔

I think this is a great solution - replacing all instance of 'currentcolor' with the current color before sending to `css-to-react-native`!

File: src/handler/expand.ts [hunk 1]
```diff
@@ -221,14 +221,15 @@ export default function expand(
       }
 
       const name = getPropertyName(prop)
+      const value = preprocess(style[prop], currentColor)
```
Comment (jueungrace) [inline src/handler/expand.ts:224]: super super nit, feel free to resolve: if we're aware of all the different properties that take currentcolor we can avoid invoking preprocess on every single property!

File: test/color-models.test.tsx [hunk 2]
```diff
@@ -219,30 +219,53 @@ describe('Color Models', () => {
...
-    //     </div>,
-    //     {
-    //       width: 100,
-    //       height: 100,
-    //       fonts,
-    //     }
-    //   )
-    //   expect(toImage(svg, 100)).toMatchImageSnapshot()
-    // })
+    it('should support currentcolor when background', async () => {
```
Comment (jueungrace) [inline test/color-models.test.tsx:222]: another smol nit: I think a more valuable test here would be to show how your work can handle nested elements - that `currentcolor` works within an element itself also well as when it's nested under a parent!

**Review by jueungrace — APPROVED** [review]
left smol comments but LGTM!

File: test/color-models.test.tsx (hunk 2 above)
Comment (LuciNyan) [reply test/color-models.test.tsx:222]: Great! Thx for your suggestions! XD

File: src/handler/expand.ts (hunk 1 above)
Comment (LuciNyan) [reply src/handler/expand.ts:224]: I have the same thought! But I'm worried that I won't be able to enumerate all properties. I'm wondering if we can fix this with a separate PR later.

**Comment by jueungrace** [comment]
@LuciNyan ah, a quick question! What are your thoughts on including support for `currentcolor` in SVG fill/stroke/etc in this PR? If not, happy to start another PR to look into it :)

**Review by shuding — APPROVED** [review]

**Pull request merged**